### PR TITLE
16 packages from framagit.org/zoggy/stog/-/archive/0.20.0/stog-0.20.0.tar.gz

### PR DESCRIPTION
**Pull request by zoggy**
This pull-request concerns:
-`stog.0.20.0`: Static web site compiler, able to handle blog posts as well as
 regular pages or any XML document in general
-`stog_all.0.20.0`: Virtual package to install all Stog libraries, tools and
 plugins
-`stog_asy.0.20.0`: Stog plugin to include Asymptote results in documents
-`stog_dot.0.20.0`: Stog plugin to generate and include graphviz graphs in
 documents
-`stog_extern.0.20.0`: Stog plugin to pipe documents in external commands
-`stog_markdown.0.20.0`: Stog plugin to use markdown syntax
-`stog_multi_doc.0.20.0`: Stog plugin to define various documents in one file
-`stog_nocaml.0.20.0`: Stog plugin to block commands executing ocaml code
-`stog_noexec.0.20.0`: Stog plugin to prevent running command with <exec>
-`stog_plugins.0.20.0`: Virtual package to install all Stog plugins
-`stog_rdf.0.20.0`: Stog plugin to generate rdf triples and execute sparql
 queries
-`stog_rel_href.0.20.0`: Stog plugin to generate relative urls
-`stog_server.0.20.0`: Stog server library
-`stog_server_multi.0.20.0`: Stog multi server library
-`stog_sitemap.0.20.0`: Stog plugin to generate a sitemap file
-`stog_writing.0.20.0`: Stog plugin to generate table of contents and
 bibliographies

---
* Homepage: https://www.good-eris.net/stog/
* Source repo: git+https://framagit.org/zoggy/stog.git
* Bug tracker: https://framagit.org/zoggy/stog/issues

---
:camel: Pull-request generated by opam-publish v2.1.0

### CODE REVIEW

**Comment by mseri** [comment]
They all fail with
```
install".

#=== ERROR while compiling stog_asy.0.20.0 ====================================#
# context              2.0.10 | linux/x86_64 | ocaml-base-compiler.4.02.3 | pinned(https://framagit.org/zoggy/stog/-/archive/0.20.0/stog-0.20.0.tar.gz)
# path                 ~/.opam/4.02/.opam-switch/build/stog_asy.0.20.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p stog_asy -j 31 --promote-install-files=false @install
# exit-code            1
# env-file             ~/.opam/log/stog_asy-2301-e7966f.env
# output-file          ~/.opam/log/stog_asy-2301-e7966f.out
### output ###
# File "plugins/dune", line 5, characters 18-25:
# 5 |  (preprocess (pps ocf_ppx))
#                       ^^^^^^^
# Error: Library "ocf_ppx" not found.
# Hint: try:
#   dune external-lib-deps --missing --no-config --root . --ignore-promoted-rules --default-target @install --always-show-command-line --promote-install-files false --promote-install-files --release --only-packages stog_asy -p stog_asy --profile release -j 31 @install
```

**Comment by mseri** [comment]
Is it expected that plugins are always conpiled, independently of the package?

**Comment by zoggy** [comment]
Indeed a dependency to ocf_ppx was missing for stog (and transitively the other pacakges will require it).
Regarding the plugins, they should depend on stog. I'm not sure to understand what you mean.

**Comment by mseri** [comment]
To be more explicit, I am suggesting to add the approprate
```
(package stog-?)
```
to all the library stanzas in https://framagit.org/zoggy/stog/-/blob/master/plugins/dune

**Comment by zoggy** [comment]
In fact the problem was that each plugin package must depend on stog. This is fixed in the new archive.

File: packages/stog_server/stog_server.0.20.0/opam [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+doc: "https://www.good-eris.net/stog/doc.html"
+bug-reports: "https://framagit.org/zoggy/stog/issues"
+depends: [
+  "dune" {>= "2.9"}
+  "stog" {= version}
+  "cryptokit" {>= "1.16.1"}
+  "ppx_deriving_yojson" {>= "3.6.1"}
+  "ojs_base_all" {>= "0.7.0"}
+  "websocket" {>= "2.14"}
+  "xmldiff" {>= "0.7.0"}
```
Comment (mseri) [inline packages/stog_server/stog_server.0.20.0/opam:16]:
```suggestion
  "xmldiff" {>= "0.7.0"}
  "xmldiff_js" {>= "0.7.0"}
```

Comment (mseri) [reply packages/stog_server/stog_server.0.20.0/opam:16]: Due to
```

#=== ERROR while compiling stog_server.0.20.0 =================================#
# context              2.0.10 | linux/x86_64 | ocaml-base-compiler.4.12.1 | file:///home/opam/opam-repository
# path                 ~/.opam/4.12/.opam-switch/build/stog_server.0.20.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p stog_server -j 31 --promote-install-files=false @install
# exit-code            1
# env-file             ~/.opam/log/stog_server-195-8032c0.env
# output-file          ~/.opam/log/stog_server-195-8032c0.out
### output ###
# File "server/dune", line 29, characters 25-35:
# 29 |   (libraries js_of_ocaml xmldiff_js yojson xtmpl_js ojs_base.js stog_server_types)
#                               ^^^^^^^^^^
# Error: Library "xmldiff_js" not found.
# Hint: try:
#   dune external-lib-deps --missing --no-config --root . --ignore-promoted-rules --default-target @install --always-show-command-line --promote-install-files false --promote-install-files --release --only-packages stog_server -p stog_server --profile release -j 31 @install
```

Comment (zoggy) [reply packages/stog_server/stog_server.0.20.0/opam:16]: Indeed, thanks.

File: packages/stog/stog.0.20.0/opam [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  "Static web site compiler, able to handle blog posts as well as regular pages or any XML document in general"
+maintainer: "zoggy@bat8.org"
+authors: "Zoggy <zoggy@bat8.org>"
+license: "GPL-3.0-only"
+homepage: "https://www.good-eris.net/stog/"
+doc: "https://www.good-eris.net/stog/doc.html"
+bug-reports: "https://framagit.org/zoggy/stog/issues"
+depends: [
+  "dune" {>= "2.9"}
+  "ocaml" {>= "4.12.0" & < "4.13"}
```
Comment (kit-ty-kate) [inline packages/stog/stog.0.20.0/opam:12]: Just curious, why is it not compatible with 4.13?

Comment (zoggy) [reply packages/stog/stog.0.20.0/opam:12]: The CI logs has been removed now that the package is ok, but IIRC this is due to a change in OCaml's internals, called from the stog part embedding the toplevel.

Comment (kit-ty-kate) [reply packages/stog/stog.0.20.0/opam:12]: Ah, you’re using `Toploop`. I see, thanks.

**Comment by kit-ty-kate** [comment]
Thanks!

**Pull request merged**